pulpcore/app/models/publication.py

```
@@ -69,6 +70,29 @@ def create(cls, repository_version, pass_through=False):
             Adds a Task.created_resource for the publication.
         """
         with transaction.atomic():
+            # Check if at least some remote artifacts have allowed checksum
+            for ca in ContentArtifact.objects.filter(
+                content__in=repository_version.content, artifact=None
+            ):
+                allowed_checksum_content = False
+                remote_artifacts = ca.remoteartifact_set.all()
+                for checksum in Artifact.DIGEST_FIELDS:
+                    if any([ra for ra in remote_artifacts if getattr(ra, checksum, None)]):
+                        allowed_checksum_content = True
+                        break
+                # We don't need to continue, allowed checksum type found
```

we do need to continue here because we need to perform this check for every CA within the repo_version if you are going to break on line 85 you will end up checking only first CA that the repo version contains

---

we should instead check the value of `allowed_checksum_content` when we're out of the forloop, false value will point that none of the remote artifacts from the ca contains allowed checksum and we should raise an error.